--- a/net/minecraft/world/level/material/LavaFluid.java
+++ b/net/minecraft/world/level/material/LavaFluid.java
@@ -123,7 +_,7 @@
 
     private boolean hasFlammableNeighbours(LevelReader p_76228_, BlockPos p_76229_) {
         for (Direction direction : Direction.values()) {
-            if (this.isFlammable(p_76228_, p_76229_.relative(direction))) {
+            if (this.isFlammable(p_76228_, p_76229_.relative(direction), direction.getOpposite())) {
                 return true;
             }
         }
@@ -205,7 +_,7 @@
             FluidState fluidstate = p_76220_.getFluidState(p_76221_);
             if (this.is(FluidTags.LAVA) && fluidstate.is(FluidTags.WATER)) {
                 if (p_76222_.getBlock() instanceof LiquidBlock) {
-                    p_76220_.setBlock(p_76221_, Blocks.STONE.defaultBlockState(), 3);
+                    p_76220_.setBlock(p_76221_, net.neoforged.neoforge.event.EventHooks.fireFluidPlaceBlockEvent(p_76220_, p_76221_, p_76221_, Blocks.STONE.defaultBlockState()), 3);
                 }
 
                 this.fizz(p_76220_, p_76221_);
