--- a/net/minecraft/client/gui/components/ChatComponent.java
+++ b/net/minecraft/client/gui/components/ChatComponent.java
@@ -357,7 +_,7 @@
         return getHeight(this.isChatFocused() ? this.minecraft.options.chatHeightFocused().get() : this.minecraft.options.chatHeightUnfocused().get());
     }
 
-    private double getScale() {
+    public double getScale() {
         return this.minecraft.options.chatScale().get();
     }
 
