--- a/net/minecraft/client/KeyboardHandler.java
+++ b/net/minecraft/client/KeyboardHandler.java
@@ -596,6 +_,7 @@
                     }
                 }
             }
+            net.neoforged.neoforge.client.ClientHooks.onKeyInput(p_446050_, p_90895_);
         }
     }
 
@@ -604,7 +_,9 @@
             Screen screen = this.minecraft.screen;
             if (screen != null && this.minecraft.getOverlay() == null) {
                 try {
-                    screen.charTyped(p_445415_);
+                    if (net.neoforged.neoforge.client.ClientHooks.onScreenCharTypedPre(screen, p_445415_)) return;
+                    if (screen.charTyped(p_445415_)) return;
+                    net.neoforged.neoforge.client.ClientHooks.onScreenCharTypedPost(screen, p_445415_);
                 } catch (Throwable throwable) {
                     CrashReport crashreport = CrashReport.forThrowable(throwable, "charTyped event handler");
                     screen.fillCrashDetails(crashreport);
