--- a/net/minecraft/client/renderer/item/BlockModelWrapper.java
+++ b/net/minecraft/client/renderer/item/BlockModelWrapper.java
@@ -175,7 +_,9 @@
             TextureSlots textureslots = resolvedmodel.getTopTextureSlots();
             List<BakedQuad> list = resolvedmodel.bakeTopGeometry(textureslots, modelbaker, BlockModelRotation.IDENTITY).getAll();
             ModelRenderProperties modelrenderproperties = ModelRenderProperties.fromResolvedModel(modelbaker, resolvedmodel, textureslots);
-            Function<ItemStack, RenderType> function = BlockModelWrapper.detectRenderType(list);
+            var renderTypeGroup = resolvedmodel.getTopAdditionalProperties().getOptional(
+                    net.neoforged.neoforge.client.model.NeoForgeModelProperties.RENDER_TYPE);
+            Function<ItemStack, RenderType> function = renderTypeGroup == null ? BlockModelWrapper.detectRenderType(list) : ignored -> renderTypeGroup.entity();
             return new BlockModelWrapper(this.tints, list, modelrenderproperties, function);
         }
 
