--- a/net/minecraft/client/renderer/rendertype/RenderType.java
+++ b/net/minecraft/client/renderer/rendertype/RenderType.java
@@ -24,13 +_,13 @@
 
 @OnlyIn(Dist.CLIENT)
 public class RenderType {
-    public static final int MEGABYTE = 1048576;
+    private static final int MEGABYTE = 1048576;
     public static final int BIG_BUFFER_SIZE = 4194304;
     public static final int SMALL_BUFFER_SIZE = 786432;
     public static final int TRANSIENT_BUFFER_SIZE = 1536;
-    public final RenderSetup state;
-    public final Optional<RenderType> outline;
-    public final String name;
+    private final RenderSetup state;
+    private final Optional<RenderType> outline;
+    protected final String name;
 
     private RenderType(String p_459041_, RenderSetup p_459042_) {
         this.name = p_459041_;
@@ -40,7 +_,7 @@
             : Optional.empty();
     }
 
-    static RenderType create(String p_459033_, RenderSetup p_458928_) {
+    public static RenderType create(String p_459033_, RenderSetup p_458928_) {
         return new RenderType(p_459033_, p_458928_);
     }
 
