--- a/net/minecraft/world/level/storage/loot/LootContext.java
+++ b/net/minecraft/world/level/storage/loot/LootContext.java
@@ -130,6 +_,11 @@
             return this;
         }
 
+        public LootContext.Builder withQueriedLootTableId(Identifier queriedLootTableId) {
+            this.queriedLootTableId = queriedLootTableId;
+            return this;
+        }
+
         public ServerLevel getLevel() {
             return this.params.getLevel();
         }
@@ -140,7 +_,7 @@
             RandomSource randomsource = Optional.ofNullable(this.random)
                 .or(() -> p_298622_.map(serverlevel::getRandomSequence))
                 .orElseGet(serverlevel::getRandom);
-            return new LootContext(this.params, randomsource, minecraftserver.reloadableRegistries().lookup());
+            return new LootContext(this.params, randomsource, minecraftserver.reloadableRegistries().lookup(), queriedLootTableId);
         }
     }
 
