--- a/com/mojang/blaze3d/opengl/GlTexture.java
+++ b/com/mojang/blaze3d/opengl/GlTexture.java
@@ -47,9 +_,10 @@
 
     public int getFbo(DirectStateAccess p_412345_, @Nullable GpuTexture p_410364_) {
         int i = p_410364_ == null ? 0 : ((GlTexture)p_410364_).id;
+        var useStencil = p_410364_ != null && p_410364_.getFormat().hasStencilAspect();
         return this.fboCache.computeIfAbsent(i, p_411998_ -> {
             int j = p_412345_.createFrameBufferObject();
-            p_412345_.bindFrameBufferTextures(j, this.id, i, 0, 0);
+            p_412345_.bindFrameBufferTextures(j, this.id, i, 0, 0, useStencil);
             return j;
         });
     }
