--- a/net/minecraft/client/renderer/ItemBlockRenderTypes.java
+++ b/net/minecraft/client/renderer/ItemBlockRenderTypes.java
@@ -19,6 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class ItemBlockRenderTypes {
+    @Deprecated
     private static final Map<Block, ChunkSectionLayer> TYPE_BY_BLOCK = Util.make(Maps.newHashMap(), p_458800_ -> {
         ChunkSectionLayer chunksectionlayer = ChunkSectionLayer.TRIPWIRE;
         p_458800_.put(Blocks.TRIPWIRE, chunksectionlayer);
@@ -362,6 +_,8 @@
     });
     private static boolean cutoutLeaves;
 
+    /** @deprecated Neo: Use {@link net.minecraft.client.renderer.block.model.BlockModelPart#getRenderType(BlockState)}. */
+    @Deprecated // Note: this method does NOT support model-based render types
     public static ChunkSectionLayer getChunkRenderType(BlockState p_109283_) {
         Block block = p_109283_.getBlock();
         if (block instanceof LeavesBlock) {
@@ -372,6 +_,8 @@
         }
     }
 
+    /** @deprecated Neo: Use {@link net.neoforged.neoforge.client.RenderTypeHelper#getMovingBlockRenderType(ChunkSectionLayer)} with the result of {@link net.minecraft.client.renderer.block.model.BlockModelPart#getRenderType(BlockState)}. */
+    @Deprecated // Note: this method does NOT support model-based render types
     public static RenderType getMovingBlockRenderType(BlockState p_109294_) {
         Block block = p_109294_.getBlock();
         if (block instanceof LeavesBlock) {
